From 6bf6f8a895381a021430d193eef3565cd50733cf Mon Sep 17 00:00:00 2001
From: Justin Song <flyingecar@gmail.com>
Date: Thu, 24 Oct 2019 12:27:14 +0200
Subject: [PATCH 581/587] ALSA: usb-audio: Add DSD support for Gustard U16/X26
 USB Interface

commit e2995b95a914bbc6b5352be27d5d5f33ec802d2c upstream.

This patch adds native DSD support for Gustard U16/X26 USB Interface.
Tested using VID and fp->dsd_raw method.

Signed-off-by: Justin Song <flyingecar@gmail.com>
Cc: <stable@vger.kernel.org>
Link: https://lore.kernel.org/r/CA+9XP1ipsFn+r3bCBKRinQv-JrJ+EHOGBdZWZoMwxFv0R8Y1MQ@mail.gmail.com
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 sound/usb/quirks.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index b6f7b13768a1..4115916c3834 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -1654,6 +1654,7 @@ u64 snd_usb_interface_dsd_format_quirks(struct snd_usb_audio *chip,
 	case 0x20b1:  /* XMOS based devices */
 	case 0x152a:  /* Thesycon devices */
 	case 0x25ce:  /* Mytek devices */
+	case 0x292b:  /* Gustard/Ess based devices */
 	case 0x2ab6:  /* T+A devices */
 	case 0x3842:  /* EVGA */
 	case 0xc502:  /* HiBy devices */
-- 
2.17.1

